Controller/MediaController.php

```diff
@@ -44,27 +45,28 @@ public function getMedia($id)
     /**
      * @throws NotFoundHttpException
      *
-     * @param string $id
-     * @param string $format
+     * @param Request $request
+     * @param string  $id
+     * @param string  $format
      *
      * @return Response
      */
-    public function downloadAction($id, $format = 'reference')
+    public function downloadAction(Request $request, $id, $format = 'reference')
```

Comment:
I did update this PR with a note on UPGRADE-4.0 and a changelog. If we find a way to do this in 3.x, I will create another PR first for 3.x branch